test/extensions/filters/listener/proxy_protocol/proxy_protocol_test.cc

```diff
@@ -230,6 +232,42 @@ TEST_P(ProxyProtocolTest, V1Basic) {
   disconnect();
 }
 
+TEST_P(ProxyProtocolTest, AllowTinyNoProxyProtocol) {
+  // allows a small request (less bytes than v1/v2 signature) through even though it doesn't use
+  // proxy protocol
+  envoy::extensions::filters::listener::proxy_protocol::v3::ProxyProtocol proto_config;
+  proto_config.set_allow_requests_without_proxy_protocol(true);
+  connect(true, &proto_config);
+
+  std::string msg = "data";
+  EXPECT_GT(PROXY_PROTO_V1_SIGNATURE_LEN, msg.length());
+  EXPECT_GT(PROXY_PROTO_V2_SIGNATURE_LEN, msg.length());
+
+  write(msg);
+
+  expectData(msg);
+
+  disconnect();
+}
+
+TEST_P(ProxyProtocolTest, AllowLargeNoProxyProtocol) {
+  // allows a large request (more bytes than v1/v2 signature) through even though it doesn't use
+  // proxy protocol
+  envoy::extensions::filters::listener::proxy_protocol::v3::ProxyProtocol proto_config;
+  proto_config.set_allow_requests_without_proxy_protocol(true);
+  connect(true, &proto_config);
+
+  std::string msg = "more data more data more data";
```

Comment:
Maybe we can test the v2 case here, otherwise, I didn't see a difference between the previous test case, both of them are going to skip the header in the first bytes.

Reply:
I don't think we can, this test is for separate logic (it ensures we have [this check](https://github.com/envoyproxy/envoy/blob/166746107b979696e7c44b914c897db19e01c442/source/extensions/filters/listener/proxy_protocol/proxy_protocol.cc#L414-L419)) while [the test prior](https://github.com/envoyproxy/envoy/blob/166746107b979696e7c44b914c897db19e01c442/test/extensions/filters/listener/proxy_protocol/proxy_protocol_test.cc#L235) ensures that we have [this check](https://github.com/envoyproxy/envoy/blob/166746107b979696e7c44b914c897db19e01c442/source/extensions/filters/listener/proxy_protocol/proxy_protocol.cc#L469-L474)